docs/src/main/asciidoc/firestore.adoc

```diff
@@ -110,6 +110,7 @@ include::{project-root}/spring-cloud-gcp-data-firestore/src/test/java/org/spring
 This annotation is optional, by default the collection name is derived from the class name.
 
 `@DocumentId` annotation marks a field to be used as document id. This annotation is required.
+NOTE: If the property annotated with `@DocumentId` is `null` and the property type is `String`, the document id is generated automatically when the entity is saved.
```

Comment:
@dmitry-s Could the id field be an `Integer`? In that case, what happens if it's `null`?
It just seems odd that it's so specific to `String`.

Reply:
It can't be an `Integer`. The client library can generate `String` ids only.

Reply:
Well, _we're_ actually generating the IDs, but yes, because they need to be very random, it only works for strings.

I would maybe clarifying by adding that ID auto-generation does not work for types other than `String`.

Reply:
Actually we only support `String` ids. I'll document this.